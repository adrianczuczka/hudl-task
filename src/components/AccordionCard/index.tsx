import React, { EventHandler, FC, ReactElement, SyntheticEvent } from 'react';
import Accordion from 'react-bootstrap/Accordion';
import Card from 'react-bootstrap/Card';
import Image from 'react-bootstrap/cjs/Image';

import styles from './styles.module.scss';

type Props = {
  title: string;
  children?: string | ReactElement;
  onClick?: EventHandler<SyntheticEvent>;
  image?: string;
};

const AccordionCard: FC<Props> = ({ title, children, onClick, image }: Props) => {
  return (
    <Card>
      <Accordion.Toggle
        as={Card.Header}
        eventKey={title}
        onClick={onClick}
        className={styles.toggle}
      >
        {title}
        {image && <Image src={image} className={styles.image} />}
      </Accordion.Toggle>
      <Accordion.Collapse eventKey={title}>
        <Card.Body>{children}</Card.Body>
      </Accordion.Collapse>
    </Card>
  );
};

export default AccordionCard;
