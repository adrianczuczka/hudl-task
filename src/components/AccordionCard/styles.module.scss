.toggle {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.image {
  width: 50px;
  height: 70px;
}
