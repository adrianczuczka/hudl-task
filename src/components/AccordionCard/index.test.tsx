import React from 'react';
import renderer from 'react-test-renderer';
import AccordionCard from './index';

it('renders correctly with only title', () => {
  const tree = renderer.create(<AccordionCard title="Test" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with title and children', () => {
  const tree = renderer.create(<AccordionCard title="Test">Test Child</AccordionCard>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with title, children, and onClick', () => {
  const tree = renderer
    .create(
      <AccordionCard
        title="Test"
        onClick={() => {
          console.log('Test');
        }}
      >
        Test Child
      </AccordionCard>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
