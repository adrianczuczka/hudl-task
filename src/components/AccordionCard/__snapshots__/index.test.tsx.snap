// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with only title 1`] = `
<div
  className="card"
>
  <div
    className="card-header"
    onClick={[Function]}
  >
    Test
  </div>
  <div
    aria-expanded={null}
    className="collapse"
  >
    <div
      className="card-body"
    />
  </div>
</div>
`;

exports[`renders correctly with title and children 1`] = `
<div
  className="card"
>
  <div
    className="card-header"
    onClick={[Function]}
  >
    Test
  </div>
  <div
    aria-expanded={null}
    className="collapse"
  >
    <div
      className="card-body"
    >
      Test Child
    </div>
  </div>
</div>
`;

exports[`renders correctly with title, children, and onClick 1`] = `
<div
  className="card"
>
  <div
    className="card-header"
    onClick={[Function]}
  >
    Test
  </div>
  <div
    aria-expanded={null}
    className="collapse"
  >
    <div
      className="card-body"
    >
      Test Child
    </div>
  </div>
</div>
`;
